components/calendar/benches/convert.rs

```diff
@@ -0,0 +1,79 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+mod fixtures;
+
+use criterion::{
+    black_box, criterion_group, criterion_main, measurement::WallTime, BenchmarkGroup, Criterion,
+};
+use icu_calendar::{AsCalendar, Calendar, Date};
+
+fn bench_date<A: AsCalendar>(date: Date<A>) {
+    // Retrieving vals
+    let _ = black_box(date.year().number);
+    let _ = black_box(date.month().ordinal);
+    let _ = black_box(date.day_of_month().0);
+}
+
+fn bench_calendar<C: Copy + Calendar>(
+    group: &mut BenchmarkGroup<WallTime>,
+    name: &str,
+    calendar: C,
+) {
+    group.bench_function(name, |b| {
+        b.iter(|| {
+            let iso = Date::try_new_iso_date(1970, 1, 2).unwrap();
+            let converted = iso.to_calendar(calendar);
+            black_box(bench_date(converted));
+        })
+    });
```

Comment:
Nit: You want the `black_box` around constant data. Something like this would be better:

```suggestion
    let iso = Date::try_new_iso_date(1970, 1, 2).unwrap();
    group.bench_function(name, |b| {
        b.iter(|| {
            let converted = black_box(iso).to_calendar(calendar);
            bench_date(converted);
        })
    });
```

Also I would probably inline `bench_date` into this function because this is the only call site.

Then you can return something like the sum of these values from the iter function.

Reply:
Done